### PR TITLE
Document monadic operators for Decoded.

### PR DESCRIPTION
**Pull request by jdhealy**
Documents the monadic operators (`flatMap`, `map`, `apply`) for `Decoded`.

Option-clicking operators will show documentation in a popover.

graphics-books-732295

:notebook_with_decorative_cover: :notebook_with_decorative_cover:   :notebook_with_decorative_cover:

### CODE REVIEW

File: Argo/Types/Decoded.swift [hunk 1]
```diff
@@ -70,14 +70,46 @@ public func pure<A>(a: A) -> Decoded<A> {
...
+  - returns: A value of type `Decoded<B>`
+*/
 public func <^> <A, B>(f: A -> B, a: Decoded<A>) -> Decoded<B> {
   return a.map(f)
 }
 
+/**
+  apply a `Decoded` function to a `Decoded` value
+  - If the function is a failing case (`.TypeMismatch`, `.MissingKey`), this will return the function's value
+  - If the function is a `.Success` case and the value is a failure case, return the value
```
Comment (tonyd256) [inline Argo/Types/Decoded.swift]: to keep with consistency can this say `, this will return the value`

File: Argo/Types/Decoded.swift [hunk 2]
```diff
@@ -70,14 +70,46 @@ public func pure<A>(a: A) -> Decoded<A> {
...
+  - If both self and the function are `.Success`, the function will be applied to the unwrapped value
+
+  - parameter f: A Decoded transformation function from type `A` to type `B`
```
Comment (tonyd256) [inline Argo/Types/Decoded.swift]: put `Decoded` in back ticks

**Comment by tonyd256** [comment]
Looks good! thanks for doing this man. Just a few comments.

**Comment by jdhealy** [comment]
:ok_hand::question: 

:eyes: :pray:

**Comment by tonyd256** [comment]
Thanks @jdhealy!! Merged in here: e469524e153daee6accb16aea8a9502afb0c6e49

**Pull request closed without merging**